drake/core/test/matrix_compare_test.cc

```diff
@@ -0,0 +1,152 @@
+#include "drake/util/eigen_matrix_compare.h"
+#include "gtest/gtest.h"
+
+using drake::util::MatrixCompareType;
+
+namespace drake {
+namespace test {
```

Comment:
Actually, I'm now leaning towards the following to be consistent with the directory structure.

```
namespace drake {
namespace core {
namespace test {
```